src/credentials/CHIPCertToX509.cpp

```diff
@@ -0,0 +1,853 @@
+    ASN1_START_BIT_STRING_ENCAPSULATED
+    {
+        // Ecdsa-Sig-Value ::= SEQUENCE
+        ASN1_START_SEQUENCE
+        {
+            // r INTEGER
+            err = writer.PutValue(kASN1TagClass_Universal, kASN1UniversalTag_Integer, false, certData.Signature.R,
+                                  certData.Signature.RLen);
+            SuccessOrExit(err);
+
+            // s INTEGER
+            err = writer.PutValue(kASN1TagClass_Universal, kASN1UniversalTag_Integer, false, certData.Signature.S,
+                                  certData.Signature.SLen);
+            SuccessOrExit(err);
+        }
+        ASN1_END_SEQUENCE;
+    }
+    ASN1_END_ENCAPSULATED;
+
+exit:
+    return err;
+}
+
+/**
+ * @brief Decode and convert the To-Be-Signed (TBS) portion of the CHIP certificate
+ *        into X.509 DER encoded form.
+ *
+ * @param reader    A TLVReader positioned at the beginning of the TBS portion
+ *                  (certificate serial number) of the CHIP certificates.
+ * @param writer    A reference to the ASN1Writer to store DER encoded TBS portion of
+ *                  the CHIP certificate.
+ * @param certData  Structure containing data extracted from the TBS portion of the
+ *                  CHIP certificate.
+ *
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ **/
+CHIP_ERROR DecodeConvertTBSCert(TLVReader & reader, ASN1Writer & writer, ChipCertificateData & certData)
+{
+    CHIP_ERROR err;
+    uint64_t tag;
+    uint8_t sigAlgoId;
+    OID sigAlgoOID;
+
+    // tbsCertificate TBSCertificate,
+    // TBSCertificate ::= SEQUENCE
+    ASN1_START_SEQUENCE
+    {
+        // version [0] EXPLICIT Version DEFAULT v1
+        ASN1_START_CONSTRUCTED(kASN1TagClass_ContextSpecific, 0)
+        {
+            // Version ::= INTEGER { v1(0), v2(1), v3(2) }
+            ASN1_ENCODE_INTEGER(2);
+        }
+        ASN1_END_CONSTRUCTED;
+
+        err = reader.Next(kTLVType_ByteString, ContextTag(kTag_SerialNumber));
+        SuccessOrExit(err);
+
+        // serialNumber CertificateSerialNumber
+        // CertificateSerialNumber ::= INTEGER
+        err = writer.PutValue(kASN1TagClass_Universal, kASN1UniversalTag_Integer, false, reader);
+        SuccessOrExit(err);
+
+        // signature AlgorithmIdentifier
+        // AlgorithmIdentifier ::= SEQUENCE
+        ASN1_START_SEQUENCE
+        {
+            err = reader.Next(kTLVType_UnsignedInteger, ContextTag(kTag_SignatureAlgorithm));
+            SuccessOrExit(err);
+
+            err = reader.Get(sigAlgoId);
+            SuccessOrExit(err);
+
+            sigAlgoOID = GetOID(kOIDCategory_SigAlgo, sigAlgoId);
+            ASN1_ENCODE_OBJECT_ID(sigAlgoOID);
+
+            certData.SigAlgoOID = sigAlgoOID;
+        }
+        ASN1_END_SEQUENCE;
+
+        // issuer Name
+        //
+        // NOTE: Accept a core tag as well as a context tag to support early CHIP certificates where
+        // this field was encoded incorrectly.
```

Comment:
Where are these early CHIP certificates coming from?  We haven't shipped anything.  There should be no such certificates we care about.

If this is really a thing we need to do, the spec needs to say so.  But I really doubt we need to do this.